packages/amplify_authenticator/lib/src/blocs/auth/auth_bloc.dart

```diff
@@ -146,6 +146,26 @@ class StateMachineBloc {
     }
   }
 
+  Stream<AuthState> _isValidSession() async* {
+    try {
+      bool isValidSession = await _authService.isValidSession;
+      if (isValidSession) {
+        yield const Authenticated();
+      } else {
+        yield AuthFlow.signin;
+      }
+    } on AmplifyException catch (e) {
+      /// On all AmplifyExceptions other than SignedOutException or SessionExpired,
+      /// we provide a message to the user and surface the SignIn form.
+      _exceptionController.add(AuthenticatorException(e.message));
```

Comment:
Are there known exceptions that this can throw other than SignedOutException or SessionExpired?

I think this is fine for now, but I am wondering how useful this message will be to an end user, and how we will handle this when we localize error messages (assuming we will do that). When/if we do that, we will have to map amplify exceptions to specific messages. For anything that we don't have mapped we will either have to display a generic error message (ex: "An unknown issue occurred." or "We could not validate your session. Please login.") or suppress the error if we don't think it is useful to an end user.

Reply:
You're right about localization... let's go with a single catch with a generic message..